src/jit/hwintrinsicxarch.cpp

```diff
@@ -528,16 +528,16 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
                 retNode = gtNewSimdHWIntrinsicNode(retType, intrinsic, baseType, simdSize);
                 break;
             case 1:
-                argType = JITtype2varType(strip(info.compCompHnd->getArgType(sig, argList, &argClass)));
+                argType = JITtype2varTypeExact(strip(info.compCompHnd->getArgType(sig, argList, &argClass)));
```

Comment:
This is probably useless. The arg type is either `TYP_STRUCT` (and then using `JITtype2varTypeExact` here doesn't make any difference) or it's a scalar and then `getArgForHWIntrinsic` will use it via `genActualType` which will again erase the signedness of the type.

Reply:
It's the only place so-far. There are a few other intrinsics that will end up in the same boat, IIRC.

Reply:
Checking this now.

Reply:
> or it's a scalar and then getArgForHWIntrinsic will use it via genActualType which will again erase the signedness of the type

This seems to not be the case. `genActualType` is used only in asserts inside `getArgForHWIntrinsic` while scalar type is recovered from `StackEntry:val`.